reader/src/main/java/org/jline/reader/impl/LineReaderImpl.java

```diff
@@ -5006,7 +5006,10 @@ protected boolean doList(
         PostResult postResult = computePost(possible, null, null, completed);
         int lines = postResult.lines;
         int listMax = getInt(LIST_MAX, DEFAULT_LIST_MAX);
-        if (listMax > 0 && possible.size() >= listMax || lines >= size.getRows() - promptLines) {
+        int rows = size.getRows();
+        int possibleSize = possible.size();
+        if (possibleSize == 0 || rows == 0) return false;
+        if (listMax > 0 && possibleSize >= listMax || lines >= rows - promptLines) {
             if (!forSuggestion) {
                 // prompt
                 post = () -> new AttributedString(getAppName() + ": do you wish to see all " + possible.size()
```

Comment:
You should either define variable and reuse those in the whole method, or keep them inline. Here, we have another `possible.size()`...

Reply:
Well spotted, I missed that one. Integrated your suggestion just now.